yaaredis/commands/iter.py

```diff
@@ -81,18 +84,32 @@ class ClusterIterCommandMixin(IterCommandMixin):
     async def scan_iter(self, match=None, count=None,
                         type=type):  # pylint: disable=redefined-builtin
         nodes = await self.cluster_nodes()
+
+        async def iterate_node(node, queue):
+            nonlocal match, count, type, nodes
```

Comment:
I don't believe you use `nodes` in this method, so no need to reference it here.

Is having `nonlocal` used here actually necessary? I might be misremembering, but I was under the impression it was only necessary for places where you wanted to _modify_ values outside your scope. Referencing them should be just fine:
```
>>> def foo():
...   x = 1
...   def bar():
...     print(x)
...   bar()
...
>>> foo()
1
```

Reply:
ah great catch; this belonged to a variant where I was actually _increasing_ the queue size if `len(data) * len(nodes) > queue.size`.  I no longer think this is a good idea.

Thanks for the detailed review!

Reply:
> only necessary for places where you wanted to modify values outside your scope

quite right; I was over zealous with the `nonlocal` here